src/python/WMCore/Database/MongoDB.py

```diff
@@ -36,27 +37,48 @@ def __init__(self, database=None, server=None, port=None, replicaset=None,
                       if missing (TODO)
         :logger:      Logger
         """
-        self.server = server # '127.0.0.1'
-        self.port = port # 8230
+        self.server = server
+        self.port = port
         self.logger = logger
         self.mockMongoDB = mockMongoDB
         if mockMongoDB and mongomock is None:
             msg = "You are trying to mock MongoDB, but you do not have mongomock in the python path."
             self.logger.critical(msg)
             raise ImportError(msg)
+
+        # NOTE: We need to explicitely check for server availiability.
+        #       From pymongo Documentation:
+        #       """
+        #           ...
+        #           Starting with version 3.0 the :class:`MongoClient`
+        #           constructor no longer blocks while connecting to the server or
+        #           servers, and it no longer raises
+        #           :class:`~pymongo.errors.ConnectionFailure` if they are
+        #           unavailable, nor :class:`~pymongo.errors.ConfigurationError`
+        #           if the user's credentials are wrong. Instead, the constructor
+        #           returns immediately and launches the connection process on
+        #           background threads.
+        #           ...
+        #       """
         try:
             if mockMongoDB:
                 self.client = mongomock.MongoClient()
                 self.logger.info("NOTICE: MongoDB is set to use mongomock, instead of real database.")
             elif replicaset:
-                self.client = MongoClient(self.server, self.port, replicaset=replicaset, **kwargs )
+                self.client = MongoClient(host=self.server, port=self.port, replicaset=replicaset, **kwargs )
```

Comment:
That I believe is a leftover from when we were trying to build the full connection string for the server Url ourselves and  not splitting all the parameters to be passed to the client separately. Now this is not needed, indeed. 

Something more, I now notice that  the client parameter from the documentation is actually `replicaSet` not `replicaset`, but I did check and it was properly recognized even with the typo in it. If I change it to a non existent  replicaSet it does timeout [1]. And it works because obviously all the client parameters get lower-cased at runtime: [2]. I am fixing it anyway because I'd like to have all the variables following our proper naming convention. (here and in all the relevant config files as well).  

[1]
```
$ ipython -i -- /data/tmp/WMCore.venv3/srv/WMCore/bin/adhoc-scripts/mongoInit.py -c /data/tmp/WMCore.venv3/srv/current/config/reqmgr2ms-output/config-output.py
2022-11-15 10:50:06,353:INFO:mongoInit:<module>(): Connecting to MongoDB using the following mongoDBConfig:
{'connect': True,
 'create': False,
 'database': 'msOutputDBPreProd',
 'directConnection': False,
 'logger': <Logger __main__ (INFO)>,
 'mockMongoDB': False,
 'password': '****',
 'port': None,
 'replicaset': 'cmsweb-dev',
 'server': ['mongodb-test.cern.ch:32001',
            'mongodb-test.cern.ch:32002',
            'mongodb-test.cern.ch:32003'],
 'username': '****'}
...
ServerSelectionTimeoutError: No replica set members available for replica set name "cmsweb-dev", Timeout: 30s, Topology Description: <TopologyDescription id: 637360cef3d1bc38070ce01b, topology_type: ReplicaSetNoPrimary, servers: []>
``` 

[2]
```
$ ipython -i -- /data/tmp/WMCore.venv3/srv/WMCore/bin/adhoc-scripts/mongoInit.py -c /data/tmp/WMCore.venv3/srv/current/config/reqmgr2ms-output/config-output.py
2022-11-15 10:53:25,406:INFO:mongoInit:<module>(): Connecting to MongoDB using the following mongoDBConfig:
{'connect': True,
 'create': False,
 'database': 'msOutputDBPreProd',
 'directConnection': False,
 'logger': <Logger __main__ (INFO)>,
 'mockMongoDB': False,
 'password': '****',
 'port': None,
 'replicaSet': 'cmsweb-test',
 'server': ['mongodb-test.cern.ch:32001',
            'mongodb-test.cern.ch:32002',
            'mongodb-test.cern.ch:32003'],
 'username': '****'}

In [1]: mongoClt
Out[1]: MongoClient(host=['mongodb-test.cern.ch:32002', 'mongodb-test.cern.ch:32001', 'mongodb-test.cern.ch:32003'], document_class=dict, tz_aware=False, connect=True, replicaset='cmsweb-test', directconnection=False)

In [2]: mongoClt.topology_description.replica_set_name
Out[2]: 'cmsweb-test'

```